atlasdb-client/src/main/java/com/palantir/atlasdb/util/AnnotatedCallable.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2016 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.util;
```

Comment:
`ThreadNamingCallable` is still used by `BasicSQL` and `BasicSQLUtils` both in `commons-db` which is still Java 6. Open to thoughts on better module & package to put this in as atlasdb-client & util feels a bit off, but would like to be able to target Java 7+ which `commons-db` and `atlasdb-commons` don't currently allow (filed #961 ).

Java 6 targeted modules:
```
$ a targetCompatibility | a 1.6
atlasdb-commons/build.gradle:7:targetCompatibility = 1.6
commons-annotations/build.gradle:4:targetCompatibility = 1.6
commons-api/build.gradle:4:targetCompatibility = 1.6
commons-db/build.gradle:4:targetCompatibility = 1.6
commons-executors/build.gradle:7:targetCompatibility = 1.6
commons-proxy/build.gradle:4:targetCompatibility = 1.6
lock-api/build.gradle:7:targetCompatibility = 1.6
```